controllers/controllers.go

```diff
@@ -0,0 +1,60 @@
+package controllers
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/metrics"
+	"github.com/openshift/windows-machine-config-operator/pkg/nodeconfig"
+)
+
+// instanceReconciler contains everything needed to perform actions on a Windows instance
+type instanceReconciler struct {
+	// Client is the cache client
+	client client.Client
+	log    logr.Logger
+	// k8sclientset holds the kube client that is needed for nodeconfig
+	k8sclientset *kubernetes.Clientset
+	// clusterServiceCIDR holds the cluster network service CIDR
+	clusterServiceCIDR string
+	// watchNamespace is the namespace that should be watched for configmaps
+	watchNamespace string
+	// vxlanPort is the custom VXLAN port
+	vxlanPort string
+	// signer is a signer created from the user's private key
+	signer ssh.Signer
+	// prometheusConfig stores information required to configure Prometheus
+	prometheusNodeConfig *metrics.PrometheusNodeConfig
+	// recorder to generate events
+	recorder record.EventRecorder
+}
+
+// configureInstance adds the specified instance to the cluster. if hostname is not empty, the instance's hostname will be
+// changed to the passed in value. If annotations is not nil, the node will have the specified annotations applied to
+// it.
```

Comment:
Reading this description makes me think that setting the hostname is a property of `instanceInfo` i.e. there would be a new field `setHostname bool` in `instanceInfo`. If that is set `instanceInfo.address` will be used to set the hostname. Or can `instanceInfo.address` not be equal to `hostname`? Either way `setHostname` or `hostname` belongs to `instanceInfo`. `newInstanceInfo()` can take `platform` as an input.

Reply:
Platform is also used to decide whether to set the hostname or not and I was suggesting to encapsulate all of that in the `instanceInfo` object.

Reply:
Please remind me what we decided to do regarding this.